File: code/classes/usage.m
#import <Foundation/Foundation.h>

// Usage in Objective-C
@interface Person : NSObject
@property (nonatomic, strong) NSString *name;
@property (nonatomic) int age;

- (instancetype)initWithName:(NSString *)name age:(int)age;
- (NSString *)greet;
@end

@implementation Person
- (instancetype)initWithName:(NSString *)name age:(int)age {
    self = [super init];
    if (self) {
        _name = name;
        _age = age;
    }
    return self;
}

- (NSString *)greet {
    return [NSString stringWithFormat:@"Hello, I'm %@ and I'm %d years old", self.name, self.age];
}
@end

int main() {
    Person *person = [[Person alloc] initWithName:@"John" age:30];
    NSLog(@"%@", [person greet]);
    return 0;
}

File: code/classes/usage.ex
# Usage in Elixir
defmodule Person do
  defstruct name: nil, age: nil
  
  def new(name, age) do
    %Person{name: name, age: age}
  end
  
  def greet(person) do
    "Hello, I'm #{person.name} and I'm #{person.age} years old"
  end
end

person = Person.new("John", 30)
IO.puts Person.greet(person)